src/buildenv/buildenv.cc

```diff
@@ -106,19 +106,16 @@ static void addPkg(const Path & pkgDir, int priority)
     done.insert(pkgDir);
     createLinks(pkgDir, out, priority);
     auto propagatedFN = pkgDir + "/nix-support/propagated-user-env-packages";
-    auto propagated = string{};
-    {
-        AutoCloseFD fd = open(propagatedFN.c_str(), O_RDONLY | O_CLOEXEC);
-        if (!fd) {
-            if (errno == ENOENT)
-                return;
-            throw SysError(format("opening ‘%1%’") % propagatedFN);
-        }
-        propagated = readLine(fd.get());
+    AutoCloseFD fd = open(propagatedFN.c_str(), O_RDONLY | O_CLOEXEC);
+    if (!fd) {
+        if (errno == ENOENT)
+            return;
+        throw SysError(format("opening ‘%1%’") % propagatedFN);
+    }
+    std::experimental::optional<string> propagated;
+    while ((propagated = readUntil(fd.get(), "\n "))) {
+        postponed.insert(*propagated);
     }
-    for (const auto & p : tokenizeString<std::vector<string>>(propagated, " "))
-        if (done.find(p) == done.end())
-            postponed.insert(p);
```

Comment:
Oh does `tokenizeString` conceptually take a list of deliminators (disjunction) rather than single deliminating string (conjunction)? Fine with me then.